### PR TITLE
add pathologic staging filter

### PR DESCRIPTION
**Pull request by KeeyanGhoreshi**
Adds a pathology staging filter which overrides the clinical staging filter if it is available.  The patient for the test endpoint doesn't have a pathology staging vale, so it might be hard to see it working.  I just copied the clinical staging in my local version of the Jane Bradshaw json and changed the element type to pathologic staging to test.

## Submitter:

**Running the Application**

- [x] Manually tested in Chrome (primary browser for testing)
- [x] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [ ] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [x] Branch implements task as expected
- [ ] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [ ] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- [ ] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [ ] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [ ] Document any new APIs/extension points
- [ ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [ ] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [ ] Code is commented

**Tests**

- [ ] Existing tests passed
- [ ] Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

**Comment by danlee1025** [comment]
Test failing on `TreatmentOptionsReducers.test.js`

File: src/mcode-pilot/utils/recordToProps.js [hunk 1]
```diff
@@ -56,15 +56,22 @@ export default function getProps(patient, condition) {
                 })(),
                 "reference": condition.getMostRecentHistologicalGrade()
             },
-            "stage": {
-                "display": "stage",
+            "clinical_stage": {
+                "display": "clinical stage",
```
Comment (dehall) [inline src/mcode-pilot/utils/recordToProps.js]: Does it matter if the display for the item in the UI says "clinical stage" vs "pathologic stage"? The latest designs just show "stage".  (cc: @rdingwell)

File: src/mcode-pilot/utils/recordToProps.js [hunk 2]
```diff
@@ -56,15 +56,22 @@ export default function getProps(patient, condition) {
...
                 "mcodeElement": "onco.core.TNMClinicalStageGroup",
                 // NOTE: the section is titled pathology but all the fields referenced in it are clinical, not pathologic
                 // (all sample data in the fixtures is clinical staging)
                 "valueType": "string",
                 "value": _safeGet(condition.getMostRecentClinicalStaging(), "stage"),
                 "reference": condition.getMostRecentClinicalStaging()
             },
+            "pathologic_stage": {
+                "display": "pathologic stage",
+                "mcodeElement": "onco.core.TNMCPathologicStageGroup",
```
Comment (dehall) [inline src/mcode-pilot/utils/recordToProps.js]: Also typo here, should be `TNMPathologicStageGroup`, no `C`

**Comment by mgramigna** [comment]
Looks good. Needs a rebase

**Pull request merged**